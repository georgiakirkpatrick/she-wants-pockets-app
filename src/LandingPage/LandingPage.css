.LandingPage__hero button {
    font-size: 18px;
    height: 50px;
    width: 60%;
    background-color: rgba(255, 255, 255, 0);
    border-color: black;
    border-width: 2px;
    border-radius: 1px;
    font-family: 'Poppins', sans-serif;
    text-transform: uppercase;
    font-size: 16px;
    font-weight: bold;
}

.LandingPage h1 {
    font-size: 20px;
}

.LandingPage__hero {
    position: relative;
    height: 90vh
}

.LandingPage__hero-title {
    position: absolute;
    bottom: 5vw;
    left: 5vw;
    width: 90vw;
    background-color: rgba(255, 255, 255, 0.315);
    padding: 1em;
    border-radius: 1px;
}

.LandingPage__hero header {
    color: rgb(27, 27, 27);
    font-family: 'Poppins', sans-serif;
    margin: 0 0 1em;
}

.LandingPage__hero img {
    width: 100%;
    height: 90vh;
    object-fit: cover;
    overflow: hidden;
    object-position: 50% 0 ;
}
