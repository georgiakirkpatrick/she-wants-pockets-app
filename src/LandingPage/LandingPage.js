import React from 'react'
import LandingCategoryList from '../LandingCategoryList/LandingCategoryList'
import PrincipleList from '../PrincipleList/PrincipleList'
import EmailSignUp from '../EmailSignUp/EmailSignUp'
import TextBlock from '../TextBlock/TextBlock'
import ReadMoreLink from '../ReadMoreLink/ReadMoreLink'
import './LandingPage.css'

const LandingPage = () => {
    const principles = [
        {
            id: 1,
            title: 'Good design', 
            description: 'Clothing and accessories should be flattering, versatile, functional, and built to last.'
        },
        {
            id: 2,
            title: 'Transparent',
            description: 'Product listings should include information about who, where, and how products are manufactured throughout the supply chain.'
        },
        {
            id: 3,
            title: 'Ethical',
            description: 'Clothing and accessories should be made in workplaces that prioritize employee safety, living wages, and freedom of association (the right to join a union).'
        },
        {
            id: 4,
            title: 'Sustainable',
            description: 'Fashion products should be made with renewable, non-polluting resources whenever possible.  Production should minimize waste.'
        }
    ]

    const categories = [
        {
            key: 1,
            id: 4,
            picPath: 'https://cdn.shopify.com/s/files/1/0767/5207/products/9754_Analina_Blue_Depths_3480_web_1186x1186.jpg?v=1580429686',
            imgAlt: 'Analina Shirt Dress in Blue Depths',
            name: 'Dresses',
            slug: 'dresses'
        },
        {
            key: 2,
            id: 4,
            picPath: 'https://cdn.shopify.com/s/files/1/0767/5207/products/9754_Analina_Blue_Depths_3480_web_1186x1186.jpg?v=1580429686',
            imgAlt: 'Analina Shirt Dress in Blue Depths',
            name: 'Dresses',
            slug: 'dresses'
        },
        {
            key: 3,
            id: 4,
            picPath: 'https://cdn.shopify.com/s/files/1/0767/5207/products/9754_Analina_Blue_Depths_3480_web_1186x1186.jpg?v=1580429686',
            imgAlt: 'Analina Shirt Dress in Blue Depths',
            name: 'Dresses',
            slug: 'dresses'
        }
    ]

    const textBlock = (
        <p>
            <span>
                It is tough to be a responsible consumer when it comes to fashion.  
            </span>
            <br />
            <span className='bold'>
                Veronaut helps you make an informed decision.
            </span>
        </p>
    )

    const princReadMore = (
        <ReadMoreLink 
            link='/principles' 
            text='Read more about our principles'
        />
    )
    

    return (
        <section className='LandingPage'>
            <div className='LandingPage__hero'>
                <img 
                    src='https://cdn.shopify.com/s/files/1/0767/5207/products/KOWTOW_Faculty_Pant_Sand_Canvas_1456_web_1186x1186.jpg?v=1581027100' 
                    alt='KowTow Faculty Pant in Sand' 
                />

                <div className='LandingPage__hero-title'>
                    <header>
                        <h1>
                            Your guide to the best-designed clothes with upstanding supply chains.
                        </h1>
                    </header>
                </div>
            </div>
            
            <TextBlock text={textBlock} />

            <PrincipleList
                principles={principles}
                readMore={princReadMore}
            />

            <LandingCategoryList categories={categories} />

            <EmailSignUp />
        </section>
    )
}

export default LandingPage