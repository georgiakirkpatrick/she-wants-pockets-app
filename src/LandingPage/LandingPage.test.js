import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import LandingPage from './LandingPage'

describe ('LandingPage', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Router><LandingPage /></Router>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Router><LandingPage /></Router>)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})