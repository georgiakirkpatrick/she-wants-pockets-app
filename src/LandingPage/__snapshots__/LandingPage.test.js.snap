// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders the UI as expected 1`] = `
<section
  className="LandingPage"
>
  <div
    className="LandingPage__hero"
  >
    <img
      alt="KowTow Faculty Pant in Sand"
      src="https://cdn.shopify.com/s/files/1/0767/5207/products/KOWTOW_Faculty_Pant_Sand_Canvas_1456_web_1186x1186.jpg?v=1581027100"
    />
    <div
      className="LandingPage__hero-title"
    >
      <header>
        <h1>
          Your guide to the best-designed clothes with upstanding supply chains.
        </h1>
      </header>
    </div>
  </div>
  <div
    className="TextBlock"
  >
    <p>
      <span>
        It is tough to be a responsible consumer when it comes to fashion.
      </span>
      <br />
      <span
        className="bold"
      >
        Veronaut helps you make an informed decision.
      </span>
    </p>
  </div>
  <section
    className="PrincipleList"
  >
    <h2>
      Principles
    </h2>
    <ul>
      <li
        className="Principle"
      >
        <h3>
          Good design
        </h3>
        <p>
          Clothing and accessories should be flattering, versatile, functional, and built to last.
        </p>
      </li>
      <li
        className="Principle"
      >
        <h3>
          Transparent
        </h3>
        <p>
          Product listings should include information about who, where, and how products are manufactured throughout the supply chain.
        </p>
      </li>
      <li
        className="Principle"
      >
        <h3>
          Ethical
        </h3>
        <p>
          Clothing and accessories should be made in workplaces that prioritize employee safety, living wages, and freedom of association (the right to join a union).
        </p>
      </li>
      <li
        className="Principle"
      >
        <h3>
          Sustainable
        </h3>
        <p>
          Fashion products should be made with renewable, non-polluting resources whenever possible.  Production should minimize waste.
        </p>
      </li>
    </ul>
    <a
      className="ReadMoreLink"
      href="/principles"
      onClick={[Function]}
    >
      <p>
        <span>
          Read more about our principles
           
        </span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-arrow-right fa-w-14 "
          data-icon="arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </p>
    </a>
  </section>
  <section
    className="LandingCategoryList__section"
  >
    <header>
      <h2>
        Categories
      </h2>
    </header>
    <ul
      className="LandingCategoryList__list"
    >
      <li
        className="Category"
      >
        <a
          className="main-link"
          href="/category/4/dresses"
          onClick={[Function]}
        >
          <h2
            className="Category__title"
          >
            Dresses
          </h2>
        </a>
      </li>
      <li
        className="Category"
      >
        <a
          className="main-link"
          href="/category/4/dresses"
          onClick={[Function]}
        >
          <h2
            className="Category__title"
          >
            Dresses
          </h2>
        </a>
      </li>
      <li
        className="Category"
      >
        <a
          className="main-link"
          href="/category/4/dresses"
          onClick={[Function]}
        >
          <h2
            className="Category__title"
          >
            Dresses
          </h2>
        </a>
      </li>
    </ul>
    <a
      className="ReadMoreLink"
      href="/all-categories"
      onClick={[Function]}
    >
      <p>
        <span>
          More categories
           
        </span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-arrow-right fa-w-14 "
          data-icon="arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </p>
    </a>
  </section>
  <section
    className="EmailSignUp"
  >
    <header>
      <h2>
        Save your favorite products
      </h2>
    </header>
    <p>
      Create an account and save items to buy later.
    </p>
    <form>
      <fieldset>
        <label
          htmlFor="email-signup"
        >
          Email 
        </label>
        <input
          id="email-signup"
          name="email-signup"
          type="text"
        />
      </fieldset>
      <button
        className="FormButton"
        onClick={[Function]}
        type="button"
      >
        CREATE ACCOUNT
      </button>
    </form>
  </section>
</section>
`;
